Automattic/wp-calypso
# Signup: Remove site mockup

#### Changes proposed in this Pull Request

* As https://github.com/Automattic/wp-calypso/pull/59795 mentioned, the `onboarding-with-preview` flow is no longer in use. Also for the `<SiteMockups />` component. Thus, we can remove the `<SiteMockups />` component and site preview related code such `showSitePreview`, `hideSitePreview` and `isSitePreviewVisible`

#### Testing instructions



* Go to `/start`
* Ensure the flow works well after the removal



Related to https://github.com/Automattic/wp-calypso/issues/56264
